builder/buffer.go

```diff
@@ -115,11 +115,25 @@ func (b *Buffer) WriteEscape(value string) {
 }
 
 func (b Buffer) escape(table, value string) string {
+	var escaped_table string
+	if table != "" {
+		parts := strings.Split(table, ".")
+		if len(parts) != 1 {
+			for i, part := range parts {
+				part = strings.TrimSpace(part)
+				parts[i] = b.Quoter.ID(part)
+			}
+			escaped_table = strings.Join(parts, ".")
+		} else {
+			escaped_table = b.Quoter.ID(table)
+		}
+	}
+
 	if value == "*" {
 		if table == "" {
 			return value
 		}
-		return b.Quoter.ID(table) + ".*"
+		return escaped_table + ".*"
```

Comment:
hmm, maybe we should this field to `escapeCache` as well, so escape_table doesn't get recomputed everytime

Reply:
Actually, I think we store it already.
https://github.com/go-rel/sql/blob/56e8dba87f4ecc77b44cd7999da2754241314687/builder/buffer.go#L162-L169

So I just changed the order for the cache check. If it is done before the computation of the escaped table, then the key pf table/value/quoter reuses the result of escaped value.

Or did you mean that it would be beneficial just to have a separate cache for table names?

Reply:
> So I just changed the order for the cache check

this is what I mean

Reply:
btw, above line should call `escapeCache.Store(key, escapedValue)` first before returning

Reply:
Right... I think it does call the cache store before the return

Reply:
looks like it still not updated to store cache before returning? so actually the cache will never be stored

also, I think we can merge this if with below if?

Reply:
Yeah, sorry - I misunderstood you. I merged statements, and know "table.*" are cached too